app/code/Magento/InventoryApi/Api/SourceRepositoryInterface.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\InventoryApi\Api;
+
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryApi\Api\Data\SourceSearchResultsInterface;
+
+/**
+ * @api
+ */
+interface SourceRepositoryInterface
+{
+    /**
+     * Save Source data.
+     *
+     * @param SourceInterface $source
+     * @return SourceInterface
+     *
+     * @throws CouldNotSaveException
```

Comment:
next exception should be thrown 
    * @throws \Magento\Framework\Exception\InputException
     * @throws \Magento\Framework\Exception\StateException
     * @throws \Magento\Framework\Exception\CouldNotSaveException